### PR TITLE
Better install help

### PR DESCRIPTION
**Pull request by hmehlan**
- simplify Makefile and unify variable names 
- add "mysql++" parent dir in include directives (#include < **mysql++/** mysql++.h>) to not to have to include -I/usr/include/mysql++
- better install instructions for dependency packages (for misssed packages or no root rights) in new file docs/INSTALL.md
- augustus and auxprogs targets added to ./Makefile
  - **but** the _augustus_ target calls the _all_ target of src/Makefile, which builds targets like _etraining_ or _load2db_ in addition to _augustus_, while in src/Makefile there is an _augustus_ target that builds only the augustus binary - this might lead to confusion

### CODE REVIEW

File: auxprogs/bam2hints/README [hunk 1]
```diff
@@ -14,47 +14,11 @@ It should contain a line like:
...
      deb http://us.archive.ubuntu.com/ubuntu vivid main universe
 
+See docs/INSTALL.md (Bamtools section) for further details, especially for manual installation from sources or installation without root rights.
 
-Afterwards, go to the auxprogs/bam2hints directory and run 
+Afterwards, go to the auxprogs/bam2hints directory and run
 
    make
 
 This produces a bam2hints binary that is located in auxprogs/bam2hints.
```
Comment (MarioStanke) [inline auxprogs/bam2hints/README:24]:
```suggestion
```

File: common.mk [hunk 2]
```diff
@@ -8,7 +8,40 @@ AUGVERSION = 3.4.0
...
+#INCLUDE_PATH_MYSQL       := -I/usr/include -I/usr/include/mysql      # the path to mysql++ may have to be adjusted
+#LIBRARY_PATH_MYSQL       := -L/usr/lib -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib -Wl,-rpath,/usr/lib/x86_64-linux-gnu
+#INCLUDE_PATH_SQLITE      := -I/usr/include
+#LIBRARY_PATH_SQLITE      := -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib/x86_64-linux-gnu
+#INCLUDE_PATH_BAMTOOLS    := -I/usr/include/bamtools
+#LIBRARY_PATH_BAMTOOLS    := -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib/x86_64-linux-gnu
+#INCLUDE_PATH_HTSLIB      := -I/usr/include/htslib
+#LIBRARY_PATH_HTSLIB      := -L/usr/lib/x86_64-linux-gnu -Wl,-rpath,/usr/lib/x86_64-linux-gnu
+
+# alternatively add paths with header files to INCLS and paths with library files go LDFLAGS
```
Comment (MarioStanke) [inline common.mk:36]:
```suggestion
# alternatively add paths with header files to INCLS and paths with library files to LDFLAGS
```

File: src/Makefile [hunk 3]
```diff
@@ -19,113 +13,116 @@ CXX?=g++
 #    CXXFLAGS = -DDEBUG -g -ggdb -pg -DDEBUG_STATES
 #  in common.mk or the environment
 OPT ?=  -O3
-CXXFLAGS += -Wall -Wno-sign-compare -pedantic ${OPT} -std=c++11
+CXXFLAGS := -Wall -Wno-sign-compare -pedantic ${OPT} -std=c++11 ${CXXFLAGS}
+INCLS    += -I../include
+
```
Comment (MarioStanke) [inline src/Makefile:18]:
```suggestion

```

**Review by MarioStanke — APPROVED** [review]
Looks good to me and like you did a very good and useful job.

File: auxprogs/bam2wig/Makefile [hunk 4]
```diff
@@ -1,59 +1,72 @@
...
-	-$(CC) $(CFLAGS) $(CPPFLAGS) -c $^ -o $@ $(INCLUDES)
+bam2wig.o: bam2wig.c
+	-$(CC) $(CPPFLAGS) $(CFLAGS) $(INCLS) -c $^
 	@if [ ! -f $@ ]; then echo $(HTSERRHINT); exit 1; fi
 
 clean:
-	rm -f $(OBJECTS) bam2wig.o bam2wig
+	rm -f bam2wig.o bam2wig
 	rm -f ../../bin/bam2wig
 	if [ ! -z $(shell which python3) ] ; then cd ../../tests/short && ./execute_test.py --clean bam2wig ; fi
```
Comment (hmehlan) [inline auxprogs/bam2wig/Makefile:69]:
```suggestion
	if [ -n $(shell which python3) ] ; then cd ../../tests/short && ./execute_test.py --clean bam2wig ; fi
```

File: auxprogs/homGeneMapping/Makefile [hunk 5]
```diff
@@ -9,5 +9,5 @@ clean:
 	cd src; make clean
 	if [ ! -z $(shell which python3) ] ; then cd ../../tests/short && ./execute_test.py --clean homGeneMapping ; fi
```
Comment (hmehlan) [inline auxprogs/homGeneMapping/Makefile:10]:
```suggestion
	if [ -n $(shell which python3) ] ; then cd ../../tests/short && ./execute_test.py --clean homGeneMapping ; fi
```

File: docs/INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+
+        ../include/randseqaccess.hh:21:10: fatal error: mysql++/mysql++.h: No such file or directory
+        /usr/bin/ld: cannot find -lmysqlclient
+        /usr/bin/ld: cannot find -lmysqlpp
+
+- solutions
+  - switch off MySQL usage by setting `MYSQL = false` in [common.mk](../common.mk)
+  - or install package `libmysql++-dev`
+  - or as non-root user install package archives into a specified directory (beware of further recursive dependencies at link and runtime)
+    - execute (replace the archives mentioned here with the ones for your system/architecture)
```
Comment (hmehlan) [inline docs/INSTALL.md:178]:
```suggestion
    - execute (replace the archives mentioned here with the ones necessary for your system/architecture)
```

File: docs/INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,499 @@
...
+- possible error messages
+
+        bam2wig.c:12:10: fatal error: bgzf.h: No such file or directory
+        /usr/bin/ld: cannot find -lhts
+
+- solutions
+  - do not make auxiliary program **bam2wig**
+  - or install packages `samtools libhts-dev`
+  - or install from source
+    - execute (don't set ZLIB variables if zlib is installed on standard locations, set variables for bz2 and lzma if available and used)
```
Comment (hmehlan) [inline docs/INSTALL.md:404]:
```suggestion
    - execute (don't set ZLIB variables if zlib is installed on standard locations, set variables for bz2 and lzma if available and used
++      or don't set the disable options if bz2 and lzma are installed on standard locations)and used)
```

File: docs/INSTALL.md [hunk 8]
```diff
@@ -0,0 +1,499 @@
...
+#### [SAMtools](https://github.com/samtools/samtools)
+
+- possible error messages
+
+        samtools: not found
+
+- solutions
+  - install package `apt-get install samtools`
+  - or install from source
+    - execute (don't set ZLIB and htslib variables and flags if both are installed on standard locations)
```
Comment (hmehlan) [inline docs/INSTALL.md:439]:
```suggestion
    - execute (don't set ZLIB and htslib variables and flags if both are installed on standard locations, remove the --without-curses 
++      if this library is installed on standard locations)
```

**Pull request merged**